src/cargo/ops/cargo_compile.rs

```diff
@@ -109,6 +109,16 @@ pub fn compile_pkg<'a>(root_package: &Package,
         s.split(' ')
     }).map(|s| s.to_string()).collect::<Vec<String>>();
 
+    let all_features = root_package.manifest().summary().features().get("default")
+        .map_or(features.clone(), |x| {
+            if !no_default_features {
+                features.iter().chain(x.iter()).cloned().collect()
+            } else {
+                features.clone()
+            }
+        }
+    );
```

Comment:
The set of activated features for a package is actually best [learned through `Resolve`](https://github.com/rust-lang/cargo/blob/master/src/cargo/core/resolver/mod.rs#L121-L123) which should be available by the time this gets down below I believe.